### PR TITLE
Teach hubot which team owns which GOV.UK applications

### PR DESCRIPTION
**Pull request by tijmenb**
This script teaches Hubot to answer questions like "Who owns Collections Publisher?" and "Who owns publishing-api?". Made possible by https://github.com/alphagov/govuk-developer-docs/pull/328.

### CODE REVIEW

File: scripts/govuk_app_owners.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# This script teaches Hubot to answer questions like "Who owns Collections Publisher?"
+# and "Who owns publishing-api?"
+module.exports = (robot) ->
+  robot.hear /who owns (.*)\?/i, (res) ->
+    # Best guess of the intended application.
+    application = res.match[1].replace(' ', '-').toLowerCase()
+
+    robot.http("https://docs.publishing.service.gov.uk/apps/#{application}.json")
+      .get() (err, response, body) ->
+        if response.statusCode == 200
```
Comment (alecgibson) [inline scripts/govuk_app_owners.coffee:10]: What about the error case? Might be nice to provide feedback?

File: scripts/govuk_app_owners.coffee [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# This script teaches Hubot to answer questions like "Who owns Collections Publisher?"
+# and "Who owns publishing-api?"
+module.exports = (robot) ->
+  robot.hear /who owns (.*)\?/i, (res) ->
+    # Best guess of the intended application.
+    application = res.match[1].replace(' ', '-').toLowerCase()
```
Comment (alecgibson) [inline scripts/govuk_app_owners.coffee:6]: Don't a lot of our apps use underscores?

File: scripts/govuk_app_owners.coffee [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+    robot.http("https://docs.publishing.service.gov.uk/apps/#{application}.json")
+      .get() (err, response, body) ->
+        if response.statusCode == 200
+          data = JSON.parse(body)
+          res.reply "#{application} is owned by #{data.team}"
```
Comment (alecgibson) [inline scripts/govuk_app_owners.coffee:12]: (Any chance of linking to a Slack channel? I guess we probably don't have a proper directory of them do we?)

File: scripts/govuk_app_owners.coffee (hunk 2 above)
Comment (tijmenb) [reply scripts/govuk_app_owners.coffee:6]: Nope, only govuk_need_api, which is going to be deprecated:

https://docs.publishing.service.gov.uk/apps.html

File: scripts/govuk_app_owners.coffee (hunk 1 above)
Comment (tijmenb) [reply scripts/govuk_app_owners.coffee:10]: I'm thinking that there may be a lot of cases where this bot is triggered inadvertently ("who owns that moldy tea cup next to the monitor?"), which could become annoying. This way we at least don't have false positives, but it's a bit of a trade-off.

File: scripts/govuk_app_owners.coffee (hunk 3 above)
Comment (tijmenb) [reply scripts/govuk_app_owners.coffee:12]: Is exactly the reason why the docs now use the slack channel for team name 😄 

https://github.com/alphagov/govuk-developer-docs/pull/324/files

Comment (alecgibson) [reply scripts/govuk_app_owners.coffee:12]: Awesome. So can we just prepend the teamname with a hash in the reply?

Comment (tijmenb) [reply scripts/govuk_app_owners.coffee:12]: It's already in the team name. Not sure if we need markup or it autolinks, but we'll see that when deployed.

File: scripts/govuk_app_owners.coffee (hunk 1 above)
Comment (alecgibson) [reply scripts/govuk_app_owners.coffee:10]: Yeah probably a fair trade-off.

**Review by alecgibson — APPROVED** [review]

**Pull request merged**